spec/services/cloud_front_header_parser_spec.rb

```diff
@@ -14,7 +14,7 @@
     let(:ip) { '192.0.2.1' }
 
     before do
-      req.add_header('CloudFront-Viewer-Address', "#{ip}:#{port}")
+      req.add_header('HTTP_CLOUDFRONT_VIEWER_ADDRESS', "#{ip}:#{port}")
```

Comment:
I think that when using `add_header`, we do want to use the "normal" values, but when reading it in they will get normalized:

https://www.rubydoc.info/gems/rack/Rack/Request/Env#add_header-instance_method

Also, inside of Rails, we have an `ActionController::Request` right?

https://api.rubyonrails.org/v2.3/classes/ActionController/Request.html#M001828

and the examples also use "normal" headers instead of prefixing `HTTP_`

I think that the `HTTP_` is what goes into the raw "env" hash but the Rack::Request and ActionController::Request will do the case folding for us automatically

Reply:
D'oh! I think I had started to look at some of this stuff initially around ports and ended up at the Rack level, and seemingly forgot all about the ActionController/ActionDispatch part.

Requests are now [ActionDispatch::Request](https://edgeapi.rubyonrails.org/classes/ActionDispatch/Request.html). They report `Rack::Request::Env` an ancestor:

```
[18] pry(main)> ActionDispatch::Request.ancestors
=> [ActionDispatch::Flash::RequestMethods,
 ActionDispatch::Request,
 #<Module:0x00007f85c8852ed8>,
 ActionDispatch::RequestCookieMethods,
 Geocoder::Request,
 Rack::Request::Env,
 ActionDispatch::PermissionsPolicy::Request,
 ActionDispatch::ContentSecurityPolicy::Request,
 ActionDispatch::Http::URL,
 ActionDispatch::Http::FilterParameters,
 ActionDispatch::Http::Parameters,
 ActionDispatch::Http::MimeNegotiation,
 ActionDispatch::Http::Cache::Request,
 Rack::Request::Helpers,
 ActiveSupport::Dependencies::RequireDependency,
 ActiveSupport::ToJsonWithActiveSupportEncoder,
 Object,
 Mailable,
 PP::ObjectMixin,
 ActiveSupport::Tryable,
 JSON::Ext::Generator::GeneratorMethods::Object,
 ActiveSupport::ForkTracker::CoreExtPrivate,
 ActiveSupport::ForkTracker::CoreExt,
 Kernel,
 BasicObject]
```

But they aren't one:

```
[22] pry(main)> ActionDispatch::Request.is_a?(Rack::Request::Env)
=> false
```

Let me run with this and fix it up. Thanks for catching this!